test/integration/targets/yum/tasks/main.yml

```diff
@@ -23,6 +23,11 @@
     - ansible_distribution in ['RedHat', 'CentOS', 'ScientificLinux', 'Fedora']
     - ansible_python.version.major == 2
 
+- include: 'repo.yml'
+  when:
+    - ansible_distribution in ['RedHat', 'CentOS', 'ScientificLinux', 'Fedora']
+    - ansible_python.version.major == 2
```

Comment:
I know you are just copying the task from above, though why doesn't this work on py3?

Reply:
The yum module uses some of yum's python API, so my guess is that that's the reason, yum is old :)